src/Bicep.LangServer/Handlers/BicepBuildCommandHandler.cs

```diff
@@ -0,0 +1,130 @@
+        {
+            string compiledFilePath = PathHelper.GetDefaultBuildOutputPath(bicepFilePath);
+            string compiledFile = Path.GetFileName(compiledFilePath);
+
+            // If the template exists and contains bicep generator metadata, we can go ahead and replace the file.
+            // If not, we'll fail the build.
+            if (File.Exists(compiledFilePath) && !TemplateContainsBicepGeneratorMetadata(File.ReadAllText(compiledFilePath)))
+            {
+                return "Build failed. " + compiledFile + " already exists.";
+            }
+
+            CompilationContext? context = CompilationManager.GetCompilation(documentUri);
+
+            if (context is null)
+            {
+                throw new InvalidOperationException($"Unable to get compilation context");
+            }
+
+            SemanticModel semanticModel = context.Compilation.GetEntrypointSemanticModel();
+            KeyValuePair<BicepFile, IEnumerable<IDiagnostic>> diagnosticsByFile = context.Compilation.GetAllDiagnosticsByBicepFile()
+                .FirstOrDefault(x => x.Key.FileUri == documentUri.ToUri());
+
+            if (diagnosticsByFile.Value.Any())
+            {
+                return GetDiagnosticsMessage(diagnosticsByFile);
+            }
+
+            using (FileStream fileStream = new FileStream(compiledFilePath, FileMode.Create, FileAccess.ReadWrite))
+            {
+                TemplateEmitter emitter = new TemplateEmitter(semanticModel, ThisAssembly.AssemblyFileVersion);
+                EmitResult result = emitter.Emit(fileStream);
+
+                return "Build succeeded. Created file " + compiledFile;
+            }
+        }
+
+        private string GetDiagnosticsMessage(KeyValuePair<BicepFile, IEnumerable<IDiagnostic>> diagnosticsByFile)
+        {
+            StringBuilder sb = new StringBuilder();
+            sb.AppendLine("Build failed. Please fix below errors:");
+
+            IReadOnlyList<int> lineStarts = diagnosticsByFile.Key.LineStarts;
+
+            foreach (IDiagnostic diagnostic in diagnosticsByFile.Value)
+            {
+                (int line, int character) = TextCoordinateConverter.GetPosition(lineStarts, diagnostic.Span.Position);
+
+                // Build a code description link if the Uri is assigned
+                var codeDescription = diagnostic.Uri == null ? string.Empty : $" [{diagnostic.Uri.AbsoluteUri}]";
+
+                sb.AppendLine($"{diagnosticsByFile.Key.FileUri.LocalPath}({line + 1},{character + 1}) : {diagnostic.Level} {diagnostic.Code}: {diagnostic.Message}{codeDescription}");
+            }
+
+            return sb.ToString();
+        }
+
+
+        // Returns true if the template contains bicep _generator metadata, false otherwise
+        public bool TemplateContainsBicepGeneratorMetadata(string template)
+        {
+            if (!string.IsNullOrEmpty(template))
+            {
+                JToken jtoken = template.FromJson<JToken>();
```

Comment:
>FromJson [](http://example.com/codeflow?start=41&length=8)

This will throw when JSON is malformed.

Reply:
Added error handling here:
https://github.com/Azure/bicep/commit/437f01c6f30eb6074c372ff7e91b31b0b7c678e7